etcdserver/server.go

```diff
@@ -374,6 +377,8 @@ func NewServer(cfg *ServerConfig) (*EtcdServer, error) {
 			srv.compactor = compactor.NewPeriodic(h, srv.kv, srv)
 			srv.compactor.Run()
 		}
+
+		srv.authStore = auth.NewAuthStore(srv.be)
```

Comment:
when we get a snapshot, we need to swap be for authStore. search oldbe in this file and you will find related code block.